src/main/java/org/craftercms/studio/impl/v2/service/policy/validators/PathPolicyValidator.java

```diff
@@ -58,18 +60,27 @@ private void validatePermitted(HierarchicalConfiguration<?> config, Action actio
         var sourceRegex = config.getString(CONFIG_KEY_SOURCE_REGEX);
         if (!item.matches(sourceRegex)) {
             String modifiedItem = null;
+            StringBuilder reasonBuilder = new StringBuilder();
             var targetRegex = config.getString(CONFIG_KEY_TARGET_REGEX);
             if (targetRegex != null) {
                 modifiedItem = item.replaceAll(sourceRegex, targetRegex);
-
+                if (!modifiedItem.equalsIgnoreCase(item)) {
+                    addPolicyReason(reasonBuilder, format("Path is replaced from source regex '%s' to target regex '%s'.", sourceRegex, targetRegex));
```

Comment:
Let's change this to show the user the before and after. Something like "Path '%s' was transformed to '%s' per the project file name policy."

Reply:
The above can be used for all cases. No need for lowercase/uppercase.

Reply:
I updated the reason message.